### DISCUSSION TITLE
Badges for Gitea/Forgejo repositories

### DISCUSSION THREAD

**Discussion by Andre601** [badges/shields#9494] [Ideas] [answered]
Would love to display some stats on my profile at Codeberg regarding repositories I host there... Tho from what I could figure out does it seem like Shields doesn't have support for it?

I feel like it would be a great benefit to have support for repositories, users, etc. hosted on a Gitea/Forgejo powered backend, as it would increase the area on where Badges could be used here.

Following labels I can think of that could be of use:

- Issues (Open, closed, total)
- Pull requests (Open, closed, merged, total)
- Licenses
- Release (Latest, by tag, ...)

Of course, the biggest question is if those *can* be supported by the backend of Shields, which I hope they can.

Also, to support all (or most) Gitea and Forgejo instances would shields probably require to use a query parameter to define the URL.
As an example `https://img.shields.io/gitea/v/Example/Repo?gitea_url=https%3A%2F%2Fgitea.com`

**Comment by calebcartwright** [comment, marked as answer]
Some related prior discussion in #4040 and #8991

**Reply by Andre601** [reply]
Alright, thanks for the very fast response... I may look into this and if it isn't too complicated may take a chance at trying to implement such features into shields for a possible PR. Tho, given my lack in the used coding language could this take a while and/or require some feedback on certain improvements...

Either way, I'll see if there is any good aproach to contibute this.

**Reply by Andre601** [reply]
Okay, I looked into it and it sadly is way over my experience to implement :C

I hope someone with more experience will be able to implement this in the future. In the meantime am I using the dynamic badge system to display the values... Just hope the JSON path I defined is properly formatted... But I assume shields reports if path is badly formatted?

Either way, tl;dr I don't have the knowledge to make these badges a reality and hope someone else can take up this task in the future.

**Reply by calebcartwright** [reply]
> But I assume shields reports if path is badly formatted?

Lmk if I've misunderstood what you meant by "formatted", but if a json path value provided fails to work with the json path lib we use then the badge will display a corresponding error (e.g. invalid json path)
